project/src/components/cards-list/cards-list.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react';
+import Card from '../../components/card/card';
+import {Offer} from '../../types/offer';
+import {Offers} from '../../types/offer';
+import {CardClassName} from '../../const';
+
+type CardsListProps = {
+  offers: Offers;
+  cardClassName: string;
+};
+
+export default function CardsList({offers, cardClassName}: CardsListProps): JSX.Element {
+  const listClassName = () => cardClassName === CardClassName.MainPage ? `${cardClassName}__places-list` : 'near-places__list';
+  const [, setFocusedCard] = useState({} as Offer);
```

Comment:
Не следует кастовать типы без лишней на то нужды. Нам целиком оффер не нужен, только его id, но даже если использовать offer то лучше так 
```suggestion
  const [, setFocusedCard] = useState<Offer | null>(null);
```